src/sage/env.py

```diff
@@ -199,8 +199,29 @@ def var(key: str, *fallbacks: Optional[str], force: bool = False) -> Optional[st
 GRAPHS_DATA_DIR = var("GRAPHS_DATA_DIR", join(SAGE_SHARE, "graphs"))
 ELLCURVE_DATA_DIR = var("ELLCURVE_DATA_DIR", join(SAGE_SHARE, "ellcurves"))
 POLYTOPE_DATA_DIR = var("POLYTOPE_DATA_DIR", join(SAGE_SHARE, "reflexive_polytopes"))
-GAP_LIB_DIR = var("GAP_LIB_DIR", join(SAGE_LOCAL, "lib", "gap"))
-GAP_SHARE_DIR = var("GAP_SHARE_DIR", join(SAGE_SHARE, "gap"))
+
+# When the GAP SPKG is used, the corresponding variable in sage-conf
+# will be empty, so the final result will look like
+#
+#   GAP_ROOT_PATHS="$SAGE_LOCAL/lib/gap;$SAGE_SHARE/gap;"
+#
+# When the system GAP is used, on the other hand, we'll have e.g.
+#
+#   GAP_ROOT_PATHS="$SAGE_LOCAL/lib/gap;$SAGE_SHARE/gap;/usr/lib/gap"
+#
+# And both of these are correct enough. Suffixing the list with a
+# semicolon is allowed and in fact tells GAP to prepend to the list of
+# root paths rather than replace it. This is quite fine in the SPKG
+# case because we'll wind up with the same list of root paths being
+# included twice. It's also fine for the system GAP because we want
+# sage's GAP packages to take precedence (just like with executables
+# in the usual $PATH) over the system ones. In any case, the only
+# thing to watch out for here is the possibility of the empty string
+# after calling split(";").
+GAP_ROOT_PATHS = ";".join([join(SAGE_LOCAL, "lib", "gap"),
```

Comment:
I propose to store this as an array of strings (without empty ones) and not as a string with semicolons. Makes some of the logic below more easy to read.

Reply:
I'll think about this along with @tornaria's suggestion about overriding `GAP_ROOT_PATHS`. GAP itself uses a semicolon-separated string, and the semicolons have meaning beyond just separating the paths. At the beginning of the string they mean "append these paths" and at the end of the string a semicolon means "prepend these paths." I think it would be more awkward to encode that magic behavior as an empty string in a list than the current level of awkwardness.

Reply:
I moved the `GAP_ROOT_PATHS` mangling from sage.env into sage-conf and I still don't think using a list here would help, for two main reasons:

1. The semicolon appending/prepending issue I already mentioned. How to we encode a leading or trailing semicolon in a python list of strings?
2. Both the source (`./configure`, and hopefully soon, meson) and the sink (`gap -l`) for `GAP_ROOT_PATHS` use a semicolon-separated string. Neither can use a python list, so we would have to parse the string we get from the build system into a python list, and then immediately convert back to a semicolon string in sage.env. A list would be a tiny bit nicer for a human to type into `conf.py` by hand, but I think we all want to stop doing that ASAP.

Reply:
I think it's better (less confusing) that `sage.env.GAP_ROOT_PATHS` matches the environment variable `GAP_ROOT_PATHS`, which is by definition a string.